Tools/llm-tool/LLMTool.swift

```diff
@@ -193,15 +260,17 @@ struct AsyncGenerator: AsyncParsableCommand {
         abstract: "async generator"
     )
 
-    @OptionGroup var args: LLMArguments
+    @OptionGroup var args: ModelArguments
+    @OptionGroup var memory: MemoryArguments
+    @OptionGroup var generate: GenerateArguments
```

Comment:
The async version also gets these but it keeps the async generation as-is.  I am thinking of removing the async version -- I thought it would be a useful example for applications, but given how the example apps are working, I think this is just overcomplicating things and making us do twice the work to keep this in sync.

Anyway, that is for another PR :-)

Reply:
BTW the async version is removed and this refactoring is already done in #47